### ISSUE TITLE
ReactorClientHttpConnector creates new HttpClient for every request

### ISSUE THREAD

**Issue by kzander91** [spring-projects/spring-framework#33093]
In 6.1.9, since commit https://github.com/spring-projects/spring-framework/commit/7785f94c4c2b7a35d80c28f197c87c54fffc6564 (related issue: #32945), `this.httpClient` is no longer assigned. The assignment should probably have happened here:
https://github.com/spring-projects/spring-framework/blob/5356a1b1ac983c1e121d809fb54c3cb43f640f9b/spring-web/src/main/java/org/springframework/http/client/reactive/ReactorClientHttpConnector.java#L128-L131

As a consequence, a new `HttpClient` instance is created for _every_ request. Apart from the possible performance impact, this is causing a memory leak in my application: I'm using client certificate authentication configured through Boot's `SslBundle` support. This causes the creation of new `SslContext` instances alongside each new `HttpClient` in `org.springframework.boot.autoconfigure.web.reactive.function.client.ReactorClientHttpConnectorFactory`.
Then, because Netty's `SslContext` implementations don't implement `hashCode()`, Reactor Netty's connection pooling doesn't work properly, causing new HTTP connection pools to be created for every request (`config.channelHash()` returns distinct values for every request https://github.com/reactor/reactor-netty/blob/v1.1.20/reactor-netty-core/src/main/java/reactor/netty/resources/PooledConnectionProvider.java#L127-L133).
All these connection pools are then never disposed of (this is Reactor Netty's default configuration I believe), eventually causing an OOME.

---

TL;DR
* Reactor Netty creates a new HTTP connection pool for every HTTP request, because
* it includes the hash code of Netty's `SslContext` in the pool key, because
* an `SslContext` has been configured by Spring Boot's `SslBundle` feature for every request, because
* Spring Framework creates new `HttpClient` instances for every request.

The leak itself _may_ be fixed in Netty (should `SslContext` implementations have stable hash codes?), but it was surfaced by the commit in Spring Framework.

**Comment by kzander91** [comment]
Further insight: It looks like this happens to all `ReactorClientHttpConnector` instances that are _not_ Spring Beans (such as those created implicitly in `WebClient.create()` and friends), because for those instances, the `Lifecycle.start()` method isn't called.

**Comment by jhoeller** [comment]
Where does the `ReactorResourceFactory` instance come from in your scenario? It seems to not have been initialized/started at the time of `ReactorClientHttpConnector` yet, so at which point will it receive an `afterPropertiesSet()` or `start()` call?

**Comment by kzander91** [comment]
@jhoeller Like this, inside the `apply()` call:
```java
@Bean
WebClient apiV1WebClient(WebClient.Builder builder, WebClientSsl ssl) {
    return builder
            .apply(ssl.fromBundle("client-auth"))
            .build();
}
```

**Comment by jhoeller** [comment]
Alright, thanks for the deep dive! I'll fix this for 6.1.11 as suggested above then since it matches the scenario I had in mind there.
